hypothesis-python/tests/cover/test_reflection.py

```diff
@@ -693,3 +695,25 @@ def test_does_not_crash_on_utf8_lambda_without_encoding(monkeypatch):
 
     monkeypatch.setattr(reflection, "detect_encoding", None)
     assert get_pretty_function_description(is_str_pi) == "lambda x: <unknown>"
+
+
+def test_too_many_posargs_fails():
+    with pytest.raises(TypeError):
+        st.times(time.min, time.max, st.none(), st.none()).validate()
+
+
+def test_overlapping_posarg_kwarg_fails():
+    with pytest.raises(TypeError):
+        st.times(time.min, time.max, st.none(), timezones=st.just(None)).validate()
+
+
+@checks_deprecated_behaviour
+def test_fails_to_detect_kwarg_with_default_value():
+    # Unfortunately we can't detect that this is an error, so you only get
+    # the warning about passing an argument positionally.  At least it warns!
+    st.floats(0, 1, False, allow_nan=None).validate()
+
+
+def test_repr_suggests_kwargs_for_deprecated_posargs():
+    strat = st.floats(0, None, False, True)
```

Comment:
Why don't we get a deprecation warning here?

Reply:
Because of our LazyStrategy machinery (via the defines_strategy decorator), the function hasn't actually been invoked yet!

I'll add a comment to this effect.